juju/juju
# [JUJU-1654] lxd go pkg: migrate from Container to Instance APIs

The [Container methods on lxd.InstanceServer](https://github.com/lxc/lxd/blob/7e0418163fa9c114aa6dcfe3bf9eab797c376a8b/client/interfaces.go#L95) are now deprecated in favour of Instance methods. This PR changes all our uses of these Container methods to the corresponding Instance method.

## Changes
- Update `github.com/lxc/lxd` from `11ace6e` to `7e04181`. As a result, this also updates/adds the following packages:
  - `github.com/pkg/sftp`: v1.13.4 -> v1.13.5
  - `github.com/vishvananda/netlink`: v1.2.0-beta -> v1.2.1-beta.2
  - `gopkg.in/ini.v1`: v1.66.4 -> v1.66.6
  - `github.com/sirupsen/logrus`: v1.8.1 -> v1.9.0 (indirect)
  - `google.golang.org/genproto`: 6ca3db6 -> 176da50 (indirect)
  - `google.golang.org/grpc`: v1.46.2 -> v1.48.0 (indirect)
  - `gopkg.in/macaroon-bakery.v3`: v3.0.0 (**newly added**, indirect)
- Make the following substitutions in the `container/lxd` and `provider/lxd` packages:
  - `CreateContainerFromImage` -> `CreateInstanceFromImage`
  - `GetContainers()` -> `GetInstances(api.InstanceTypeContainer)`
  - `GetContainerState` -> `GetInstanceState`
  - etc...
  - `api.Container{...}` -> `api.Instance{..., Type: "container"}`
  - `api.ContainersPost` -> `api.InstancesPost`
  - `api.ContainerPut` -> `api.InstancePut`
  - `api.ContainerStatePut` -> `api.InstanceStatePut`
  - etc...
- Rename the following types:
  - `lxd.ContainerServer` -> `lxd.InstanceServer`
  - `lxdtesting.NewMockContainerServer` -> `lxdtesting.NewMockInstanceServer`
- In the Server interface in `provider/lxd/server.go`, rename the `GetContainer` and `GetContainerState` methods to `GetInstance` and `GetInstanceState` respectively.
- Regenerate the following mocks:
  - `container/lxd/testing/lxd_mock.go`
  - `provider/lxd/package_mock_test.go`
  - `upgrades/upgradevalidation/mocks/lxd_mock.go`

I have not changed any of the Juju-defined methods - only the LXD-defined ones. For example, in the Server interface in `provider/lxd/server.go`, I changed `GetContainer` and `GetContainerState` to `GetInstance` and `GetInstanceState` (since these methods are inherited from `lxd.InstanceServer` in the canonical implementation). However, I have not changed the names of the other container methods, which are defined internally by Juju.

## Checklist

*If an item is not applicable, use `~strikethrough~`.*


## QA steps

*Commands to run to verify that the change works.*

```sh
go test ./container/lxd/...
go test ./provider/lxd/...
```
